staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4613,6 +4613,14 @@ type SecurityContext struct {
 	// Default is false.
 	// +optional
 	ReadOnlyRootFilesystem *bool `json:"readOnlyRootFilesystem,omitempty" protobuf:"varint,6,opt,name=readOnlyRootFilesystem"`
+	// AllowPrivilegeEscalation controls whether a process can gain more
+	// privileges than it's parent process. This bool directly controls if
+	// the no_new_privs flag will be set on the container process.
+	// AllowPrivilegeEscalation is true always when the container is:
+	// 1) run as Privileged
+	// 2) has CAP_SYS_ADMIN
```

Comment:
@jessfraz Are we missing the case "when a container is not run as root" here?

Reply:
We got rid of that functionality

Reply:
And the proposal is outdated now? https://github.com/kubernetes/community/blob/f149433600ff1795521e7d9f2ae8ff3162394c61/contributors/design-proposals/no-new-privs.md#changes-of-securitycontext-objects

Reply:
Also some tests can be deleted then: https://github.com/kubernetes/kubernetes/blob/28f6b3fcc037c3a21048160d062f6d544081bd71/pkg/securitycontext/util_test.go#L203-L212